OrbitQt/ConnectionConfiguration.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_CONNECTION_CONFIGURATION_H_
+#define ORBIT_QT_CONNECTION_CONFIGURATION_H_
+
+#include <memory>
+#include <utility>
+
+#include "DeploymentConfigurations.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitClientData/ProcessData.h"
+#include "OrbitClientServices/ProcessManager.h"
+#include "OrbitGgp/Instance.h"
+#include "OrbitSsh/Context.h"
+#include "OrbitSsh/Credentials.h"
+#include "grpcpp/channel.h"
+#include "servicedeploymanager.h"
+
+namespace orbit_qt {
+
+struct StadiaConnection {
+ public:
+  explicit StadiaConnection(const OrbitSsh::Context* ssh_context,
+                            const ServiceDeployManager::GrpcPort& grpc_port,
+                            const DeploymentConfiguration* deployment_configuration)
+      : ssh_context(ssh_context),
+        grpc_port(grpc_port),
+        deployment_configuration(deployment_configuration) {
+    CHECK(ssh_context != nullptr);
+    CHECK(deployment_configuration != nullptr);
+  }
+  ~StadiaConnection() {
+    if (process_manager != nullptr) process_manager->Shutdown();
+  }
+  void CreateServiceDeployManager(OrbitSsh::Credentials credentials) {
+    CHECK(service_deploy_manager == nullptr);
+    service_deploy_manager = std::make_unique<ServiceDeployManager>(
+        deployment_configuration, ssh_context, std::move(credentials), grpc_port);
+  }
+
+  const OrbitSsh::Context* ssh_context;
+  const ServiceDeployManager::GrpcPort& grpc_port;
```

Comment:
When someone passes in a temporary GrpcPort as a constructor argument this will crash when CreateServiceDeployManager is called.

Reply:
(because grpc_port is a reference)

Reply:
I changed quite some things, and did a copy there. ptal if its good now, or something could still go wrong.